app/scripts/controllers/metametrics.js

```diff
@@ -944,10 +955,36 @@ export default class MetaMetricsController {
         return resolve();
       };
 
-      this.segment.track(payload, callback);
+      this._submitSegmentEvent('track', payload, callback);
       if (flushImmediately) {
         this.segment.flush();
       }
     });
   }
+
+  // Method below submits the request to analytics SDK.
+  // For MV3 it will also add event to controller store
```

Comment:
this method appears to add the event to the store whether or not it is MV3. We might want to update the code to reflect this comment

Nice comment overall! I'd like to suggest some minor tweaks:
- replace "events" to "API Calls" as mentioned in the [other comment](https://github.com/MetaMask/metamask-extension/pull/16198/files#r1007365427)
- add @param jsdoc comments
- capitalize Segment

  /**
   * Use the method below to submit API calls to Segment using the analytics SDK.
   *
   * For MV3, this method also adds the API call to the controller store and removes
   * the API call from the controller store once its callback is executed.
   * Saving API calls in the controller store enables us to ensure API calls are submitted
   * to Segment even if the service worker termintates before the API call is submitted.
   *
   * @param {string} apiCallMethod - The method name of the analytics API call to be submitted to Segment.
   *                                 e.g. identify, track, page, etc.
   * @param {object} payload
   * @param {Function} callback
   */

Reply:
@digiwand : I implemented feedbacks above except for jsdoc, I think it is not requires for internal methods.

Reply:
Thank you @jpuri 

follow-up: 
1. can we swap out `eventType` name in the parameter as well? to maybe `apiCallMethod`? 
2. can we rename "events" in the store to "apiCalls"?
3. even though the jsdoc comments aren't required, I think they can be helpful to the reader.
4. The comment still does not match the code where "For MV3 it will also add event to controller store" because we are added the api calls to the store for both MV2 and MV3 builds

Reply:
I renamed `events` to `segmentApiCalls` and also updated comments.

Renaming `eventType` to `apiCallMethod` looks confusing to me. Event before changes in this PR the field was eventType. Also I think JSDoc for internal method can create confusion.